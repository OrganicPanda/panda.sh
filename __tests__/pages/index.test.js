import React from "react";
import { mount } from "enzyme";
import IndexPage from "../../pages/index";

const useRouter = jest.spyOn(require("next/router"), "useRouter");
useRouter.mockImplementation(() => ({
  pathname: "/",
}));

describe("IndexPage", () => {
  let wrapper = null;

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
      wrapper = null;
    }
  });

  it("renders IndexPage unchanged", () => {
    wrapper = mount(<IndexPage isLoading={false} />);

    expect(wrapper).toMatchSnapshot();
  });
});
