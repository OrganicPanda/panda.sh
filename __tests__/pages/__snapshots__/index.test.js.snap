// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders IndexPage unchanged 1`] = `
<IndexPage
  isLoading={false}
>
  <Layout>
    <Head>
      <_default
        headManager={Object {}}
        inAmpMode={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <header>
      <nav>
        <NavLink
          activeClassName="navItemActive"
          href="/"
        >
          <Link
            href="/"
          >
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="navItemActive"
          href="/things"
        >
          <Link
            href="/things"
          >
            <a
              href="/things"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Things
            </a>
          </Link>
        </NavLink>
      </nav>
    </header>
    <main>
      <p>
        Hi there ðŸ‘‹
      </p>
    </main>
  </Layout>
</IndexPage>
`;
